test/http-client.spec.js

```diff
@@ -787,20 +784,49 @@ describe('http client', () => {
         });
     });
 
-    it('should be applied after all transformers', (done) => {
-      class RequestInterceptor {
+    it('should be possible to modify a header in a request interceptor', (done) => {
+      class Interceptor {
         request(message) {
+          message.headers.add('aheader', 'anothervalue');
+          return message;
+        }
+
+        response(message) {
+          var request = jasmine.Ajax.requests.mostRecent();
+          expect(request.requestHeaders['aheader']).toBe('anothervalue')
+          return message;
+        }
+      }
+
+      var interceptor = new Interceptor();
+      var client = new HttpClient()
+        .configure(x => {
+          x.withInterceptor(interceptor);
+          x.withBaseUrl(baseUrl);
+        });
+
+      client.createRequest('some/cool/path')
+        .asGet()
+        .send()
+        .then((response) => {
+          done();
+        });
+    });
+
+    it('should apply xhr transformers before response interceptors are called', (done) => {
```

Comment:
Ah. I added a test for that in the processor spec. I think most interceptor tests belong there because it allows testing at a lower level. Do you have any opinion on that?

https://github.com/alexanderchr/http-client/commit/8b51585984190b8d9e625469a87c814b8b80dd49#diff-837ec137ea7cbd265b5ac7a192bda429R176